### ISSUE TITLE
Install and configure cura

### ISSUE THREAD

**Issue by displague** [serialise/docker-octoprint#1]
Cura, the default slicer, should be installed and configured for use so .stl files can be processed out of the box.

**Comment by loa** [comment]
Have not used this so far. Would you be able to create a pull-request or just point (high level) how to configure and verify the functionality?

**Comment by displague** [comment]
I can't docker build on my pi for whatever reason... I added pybonjour because I noticed a gripe in the log.

He are the changes that I think would work:
```patch
--- Dockerfile.orig	2016-02-06 23:35:58.908023331 +0000
+++ Dockerfile	2016-02-06 23:27:46.047124035 +0000
@@ -1,11 +1,11 @@
-FROM resin/rpi-raspbian:wheezy
+FROM resin/rpi-raspbian:jessie

 RUN apt-get update && \
     apt-get install -y python python-dev python-pip python-virtualenv --no-install-recommends && \
     rm -rf /var/lib/apt/lists/*

 RUN apt-get update && \
-    apt-get install -y imagemagick libav-tools && \
+    apt-get install -y imagemagick libav-tools cura-engine && \
     rm -rf /var/lib/apt/lists/*

 ENV OCTOPRINT_DOWNLOAD_URL=https://github.com/foosel/OctoPrint/archive/dedadbc9ac0305799e94ae279d3bca131629c4c5.zip \
@@ -22,6 +22,7 @@
     rm octoprint.zip && \
     cd /srv/octoprint && \
     python setup.py install && \
+    pip install pybonjour && \
     apt-get purge -y --auto-remove $buildDeps && rm -rf /var/lib/apt/lists/*

 WORKDIR /srv/octoprint
```

**Comment by loa** [comment]
No worries! I can build the images for you _(you just have to live with the long feedback cycle)_ ;)

So I tried multiple times but I get an error while trying to install pybonjour. In my setup I have bonjour/avahi on the host system, do you get anything extra by having it installed together with octoprint?

```
Downloading/unpacking pybonjour
  Could not find any downloads that satisfy the requirement pybonjour
```

So there is a new branch issue-1 with docker image tag issue-1 that atleast contains jessie with cura-engine.

**Comment by loa** [comment]
#2

**Comment by displague** [comment]
I don't think it is necessary for cura-engine.  I just saw octoprint complaining about it in the logs within the container.

**Comment by loa** [comment]
Mine is discovered by slic3r without pybonjour and running on port 5000.

But test the issue-1 tag out and tell me if it works as expected. :)

**Comment by displague** [comment]
I'll check it out sometime this week - thanks!

**Comment by displague** [comment]
octoprint fails to start -- it looks like we need to `apt-get install python-pkg_resources` at least.  I'm trying to figure out how to run this myself to see what it takes to fix it.  Any tips?

**Comment by displague** [comment]
I got some assistance and figured out how to build the docker image (I am doing this from Ubuntu Snappy on the Pi2.. the trick was to build within the ~/app/docker directory).

It looks like we just need `python-pkg-resources` to get it started.

**Comment by displague** [comment]
`python-pip` seems to be needed to install plugins.

**Comment by displague** [comment]
python-pip runs into problems with an incompatible version of python-requests. `pip --version` fails.
  
I'm giving `easy_install -U pip requests`.

**Comment by loa** [comment]
I have installed plugins on the master branch version, so it should work?

I will start with adding the cura-engine to master :)

**Comment by displague** [comment]
The master branch was based on an older Debian.  `pip` isn't working out of the box on this version.

curaengine needs a cura config.  I don't know what this file is supposed to look like so I don't have it working yet.
